wasm/src/programs/manager/deploy.rs

```diff
@@ -204,4 +208,109 @@ impl ProgramManager {
             .saturating_mul(1_000_000); // 1 microcredit = 1e-6 credits.
         Ok(namespace_cost)
     }
+
+    /// Deploy an Aleo program
+    ///
+    /// @param private_key The private key of the sender
+    /// @param program The source code of the program being deployed
+    /// @param imports A javascript object holding the source code of any imported programs in the
+    /// form \{"program_name1": "program_source_code", "program_name2": "program_source_code", ..\}.
+    /// Note that all imported programs must be deployed on chain before the main program in order
+    /// for the deployment to succeed
+    /// @param priority_fee_credits The optional priority fee to be paid for the transaction
+    /// @param fee_record The record to spend the fee from
+    /// @param url The url of the Aleo network node to send the transaction to
+    /// @param imports (optional) Provide a list of imports to use for the program deployment in the
+    /// form of a javascript object where the keys are a string of the program name and the values
+    /// are a string representing the program source code \{ "hello.aleo": "hello.aleo source code" \}
+    /// @param fee_proving_key (optional) Provide a proving key to use for the fee execution
+    /// @param fee_verifying_key (optional) Provide a verifying key to use for the fee execution
+    /// @returns {Transaction}
+    #[wasm_bindgen(js_name = buildUpgradeTransaction)]
+    #[allow(clippy::too_many_arguments)]
+    pub async fn upgrade(
+        private_key: &PrivateKey,
+        program: &str,
+        priority_fee_credits: f64,
+        fee_record: Option<RecordPlaintext>,
+        url: Option<String>,
+        imports: Option<Object>,
+        fee_proving_key: Option<ProvingKey>,
+        fee_verifying_key: Option<VerifyingKey>,
+        offline_query: Option<OfflineQuery>,
+    ) -> Result<Transaction, String> {
+        log("Creating deployment transaction");
+        let mut process_native = ProcessNative::load_web().map_err(|err| err.to_string())?;
+        let process = &mut process_native;
+
+        log("Checking if the program is deployed on the network");
+        let node_url = url.as_deref().unwrap_or(DEFAULT_URL);
+        let program = ProgramNative::from_str(&program).map_err(|err| err.to_string())?;
+        let program_id = program.id().to_string();
+        let program_string = get_program_from_network(node_url, &program_id).await.map_err(|err| err.to_string())?;
+        let deployed_program = ProgramNative::from_str(&program_string).map_err(|err| err.to_string())?;
+        let deployed_program_edition =
+            latest_program_edition(node_url, &program.id().to_string()).await.map_err(|err| err.to_string())?;
```

Comment:
On read of this, it seems almost identical to the existing deploy function. Should we just modify deploy to run these particular checks instead of adding a new method?